src/internet_identity/src/assets.rs

```diff
@@ -134,3 +142,57 @@ fn get_assets() -> [(&'static str, &'static [u8], ContentEncoding, ContentType);
         ),
     ]
 }
+
+pub fn increase_asset_counter(asset: String, status_code: u16) {
+    const MAX_TRACKED_ASSETS: u8 = 30;
+
+    if (200..400).contains(&status_code) {
+        // prevent clients from recording successful status codes for non-existing assets
+        trap_if_not_asset(&asset);
+
+        // prevent clients from recording impossible status codes for existing assets
+        if asset == FAQ_PATH && status_code != 301 || asset != FAQ_PATH && status_code != 200 {
+            trap(&format!("Invalid status code for asset '{asset}'"));
+        }
+    }
+
+    usage_metrics_mut(|metrics| {
+        match metrics
+            .asset_loading_counters
+            .get_mut(&(asset.clone(), status_code))
+        {
+            None => {
+                if metrics.asset_loading_counters.len() >= MAX_TRACKED_ASSETS as usize {
+                    // This simple logic will cause churn on the least used asset.
```

Comment:
I'm confused. Doesn't the logic above skip tracking on unknown assets? how could we have more than `get_assets().len()` assets?

Reply:
No, it does track all status codes indicating failure, regardless if the path corresponds to an asset or not.